### PR TITLE
Feat/related projects

### PR DESCRIPTION
**Pull request by JCoronaWize**
# What does this PR do?
Implements "Add/Remove" input for Related Projects
Adds an Edit page route
Adds **_Related Projects_** section

# Where should the reviewer start?
Check the new edit page file:
`app/routes/projects/edit/$editProjectId.tsx`

# How should this be manually tested?

1. Run the application locally. Go to a project's page and click on the edit button. 
2. On the Edit page change the related projects values and submit. 
3. Go back to the project's page and check the values were saved correctly for the _**Related Projects**_ section

# Any background context you want to provide?

# What are the relevant tickets?
#52 

# Screenshots
Captura de Pantalla 2022-11-03 a la(s) 1 28 05
Captura de Pantalla 2022-11-03 a la(s) 1 28 13

# Questions
In related projects section, should the Chips components be changed to Cards, o another component? 
Could we use useFetcher for the whole edit project form?

# Checklist

- [ ] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my feature works.
- [ ] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

File: app/models/project.server.ts [hunk 1]
```diff
@@ -128,7 +165,58 @@ export async function updateProjects({
...
+  id,
+  data,
+}: {
+  id: string;
+  data: RelatedProjectInput;
+}) {
+  // Create related Projects
+  const createResponse = [];
+  let response;
+  for (let i = 0; i < data.relatedProjects.length; i++) {
```
Comment (LuisTejedaS) [inline app/models/project.server.ts:182]: Should we add all of the operations in a db transaction?

File: app/models/project.server.ts [hunk 2]
```diff
@@ -128,7 +165,58 @@ export async function updateProjects({
...
+          projectAId: id,
+          projectBId: data.relatedProjects[i].id,
+        },
+      });
+      createResponse.push(response)
+    }
+  }
+
+  // Delete related projects
+  const relatedProjectsIds = await data.relatedProjects.map((e) => e.id);
```
Comment (LuisTejedaS) [inline app/models/project.server.ts:210]: i have a doubt with this code could we set a call?

**Review by LuisTejedaS — COMMENTED** [review]
Try to resolve all TS and lint errors

File: app/models/project.server.ts (hunk 2 above)
Comment (JCoronaWize) [reply app/models/project.server.ts:210]: Sure thing, want to schedule it ?

**Comment by JCoronaWize** [comment]
Applied changes to code, and solved TS and ESLint errors

**Review by LuisTejedaS — APPROVED** [review]
Just a minor comment, it is ok if you want to merge it

**Review by LuisTejedaS — APPROVED** [review]

**Pull request merged**